src/NHibernate/Collection/Generic/PersistentGenericMap.cs

```diff
@@ -22,15 +25,22 @@ namespace NHibernate.Collection.Generic
 	[DebuggerTypeProxy(typeof(DictionaryProxy<,>))]
 	public partial class PersistentGenericMap<TKey, TValue> : AbstractPersistentCollection, IDictionary<TKey, TValue>, ICollection
 	{
-		protected IDictionary<TKey, TValue> WrappedMap;
+		protected IDictionary<TKey, TValue> WrappedMap { get; private set; }
+		private readonly ICollection<TValue> _wrappedValues;
 
-		public PersistentGenericMap() { }
+		public PersistentGenericMap()
+		{
+			_wrappedValues = new ValuesWrapper(this);
```

Comment:
x3, one for each constructor: they cannot call each over and the field cannot be initialized in place when using `this`.